conjure-codegen/src/example_types/another/test_service.rs

```diff
@@ -1192,2212 +1201,746 @@ pub trait TestService<I, O> {
+        #[path(
+            name = "branch",
+            decoder = conjure_http::server::conjure::FromPlainDecoder
+        )]
         branch: String,
     ) -> Result<Option<String>, conjure_http::private::Error>;
+    #[endpoint(
+        method = GET,
+        path = "/catalog/datasets/{datasetRid}/testParam",
+        name = "testParam",
+        produces = conjure_http::server::conjure::CollectionResponseSerializer
+    )]
     fn test_param(
         &self,
+        #[auth]
         auth_: conjure_object::BearerToken,
+        #[path(
+            name = "datasetRid",
+            decoder = conjure_http::server::conjure::FromPlainDecoder,
+            log_as = "datasetRid"
+        )]
         dataset_rid: conjure_object::ResourceIdentifier,
     ) -> Result<Option<String>, conjure_http::private::Error>;
+    #[endpoint(
+        method = POST,
+        path = "/catalog/test-query-params",
+        name = "testQueryParams",
+        produces = conjure_http::server::StdResponseSerializer
+    )]
     fn test_query_params(
         &self,
+        #[auth]
         auth_: conjure_object::BearerToken,
+        #[body(deserializer = conjure_http::server::StdRequestDeserializer)]
         query: String,
+        #[query(
+            name = "different",
+            decoder = conjure_http::server::conjure::FromPlainDecoder
+        )]
         something: conjure_object::ResourceIdentifier,
+        #[query(
+            name = "optionalMiddle",
+            decoder = conjure_http::server::conjure::FromPlainOptionDecoder,
+            log_as = "optionalMiddle"
+        )]
         optional_middle: Option<conjure_object::ResourceIdentifier>,
+        #[query(
+            name = "implicit",
+            decoder = conjure_http::server::conjure::FromPlainDecoder
+        )]
         implicit: conjure_object::ResourceIdentifier,
+        #[query(
+            name = "setEnd",
+            decoder = conjure_http::server::conjure::FromPlainSeqDecoder<_>,
+            log_as = "setEnd"
+        )]
         set_end: std::collections::BTreeSet<String>,
+        #[query(
+            name = "optionalEnd",
+            decoder = conjure_http::server::conjure::FromPlainOptionDecoder,
+            log_as = "optionalEnd"
+        )]
         optional_end: Option<conjure_object::ResourceIdentifier>,
     ) -> Result<i32, conjure_http::private::Error>;
+    #[endpoint(
+        method = POST,
+        path = "/catalog/test-no-response-query-params",
+        name = "testNoResponseQueryParams"
+    )]
     fn test_no_response_query_params(
         &self,
+        #[auth]
         auth_: conjure_object::BearerToken,
+        #[body(deserializer = conjure_http::server::StdRequestDeserializer)]
         query: String,
+        #[query(
+            name = "different",
+            decoder = conjure_http::server::conjure::FromPlainDecoder
+        )]
         something: conjure_object::ResourceIdentifier,
+        #[query(
+            name = "optionalMiddle",
```

Comment:
I assume not, but is it maybe possible to specify the rename once for the function instead of each param?

Reply:
yeah it has to be done on each param